dataquality/integrations/setfit.py

```diff
@@ -372,7 +376,11 @@ def dq_evaluate(
 
         if column_mapping is not None:
             dataset = _apply_column_mapping(dataset, column_mapping)
-
+        if "id" not in dataset.features:
+            dataset = dataset.map(lambda x, idx: {"id": idx}, with_indices=True)
+        cur_epoch = get_data_logger().logger_config.cur_epoch
+        if epoch is not None:
+            cur_epoch = epoch
```

Comment:
should this be persisted? if so,
```suggestion
        if epoch is not None:
            dq.set_epoch(epoch)
        cur_epoch = get_data_logger().logger_config.cur_epoch
```

Reply:
I thinks that's better